.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,38 @@
+name: Setup build
+description: >
+  Common repository and build tool setup
+
+inputs:
+  # note nested virtualization is not supported on current arm64 mac runners: https://github.com/runfinch/finch/issues/812
```

Comment:
nit: this issue got closed already, is it still relevant, it seems like it'll never be fixed?

Reply:
It won't be fixed but it links to all the relevant details about why we can't do what I was trying to do should we come across it again.